api/src/main/java/org/openmrs/module/initializer/api/programs/ProgramLineProcessor.java

```
@@ -47,8 +47,11 @@ public Program fill(Program program, CsvLine line) throws IllegalArgumentExcepti
 		}
 		program.setDescription(description);
 		
-		Concept outcomeConcept = Utils.fetchConcept(line.get(HEADER_OUTCOMES_CONCEPT), conceptService);
-		program.setOutcomesConcept(outcomeConcept);
+		String outcomeConceptString = line.get(HEADER_OUTCOMES_CONCEPT);
+		if (StringUtils.isNotBlank(outcomeConceptString)) {
+			Concept outcomeConcept = Utils.fetchConcept(outcomeConceptString, conceptService, true);
+			program.setOutcomesConcept(outcomeConcept);
+		}
```

Could this be sufficient:
```java
Concept outcomeConcept = Utils.fetchConcept(line.get(HEADER_OUTCOMES_CONCEPT), conceptService);
program.setOutcomesConcept(Optional.of(outcomeConcept).orElseThrow(() -> new IllegalArgumentException("...")));
```
?
In other words, could this be handled on a case by case basis in each line processor?

---

Sure, it could. Is there an example of a kind of concept for which we would not want to fail hard if the user specifies a non-existent concept?

---

Oh that's right, it's only important for nullable concepts. I don't know of other examples, off the top of my head. Perhaps drug concept? Anyway I think you make a fair point.